### ISSUE TITLE
[BUG] The policy_name field can't be resolved in multitask jobs

### ISSUE THREAD

**Issue by allebacco** [databrickslabs/dbx#149]
## Expected Behavior

`dbx` should beable to resolve `policy_name` in multitask jobs:

## Current Behavior

`dbx` is not able to resolve `policy_name` in multitask jobs with this exception:

```
File "/home/abacchini/Projects/data-solutions/data-products/dnm-lake/dnm-dsp/.venv/lib/python3.8/site-packages/dbx/commands/deploy.py", line 338, in _adjust_job_definitions
    PolicyNameProcessor(task, api_client).preprocess()
  File "/home/abacchini/Projects/data-solutions/data-products/dnm-lake/dnm-dsp/.venv/lib/python3.8/site-packages/dbx/commands/deploy.py", line 353, in preprocess
    dbx_echo(f"Processing policy name {policy_name} for job {self._job['name']}")
KeyError: 'name'
```

## Steps to Reproduce (for bugs)

The following `deployment.json` file can't be deployed.

```json
{
    "dev": {
        "jobs": [
            {
                "format": "MULTI_TASK",
                "name": "This is the name of the job",
                "tasks": [
                    {
                        "libraries": [
                            {
                                "jar": "this/is/a/jar/on/databricks.jar"
                            }
                        ],
                        "new_cluster": {
                            "aws_attributes": {
                                "availability": "SPOT_WITH_FALLBACK",
                                "ebs_volume_count": 1,
                                "ebs_volume_size": 50,
                                "ebs_volume_type": "GENERAL_PURPOSE_SSD",
                                "first_on_demand": 1
                            },
                            "node_type_id": "m5a.large",
                            "num_workers": 2,
                            "policy_name": "ThisIsMyPolicyName",
                            "spark_version": "10.2.x-scala2.12"
                        },
                        "spark_jar_task": {
                            "main_class_name": "com.camlin.arena.databricks.Main"
                        },
                        "task_key": "landing_to_bronze",
                        "timeout_seconds": 3600.0
                    }
                ]
            }
        ]
    }

```

## Context

In this example, the cluster definition is inside the definition of a task, so no `name` field is present.
The issue is with line https://github.com/databrickslabs/dbx/blob/88ecefcc6dc026add9523345ad0a2733e08b0b8d/dbx/commands/deploy.py#L353 that is accessing the job name, even if the `self._job` is a task definition object.

The PR #142 do not fix the problem for this case.

## Your Environment

* dbx version used: 0.3.1
* Databricks Runtime version: 10.2
* Jobs API 2.1

**Comment by allebacco** [comment]
Moreover, from looking at the code in 
https://github.com/databrickslabs/dbx/blob/88ecefcc6dc026add9523345ad0a2733e08b0b8d/dbx/commands/deploy.py#L316
it seems that the `policy_name` will not be resolved if clusters are defined in the `job_clusters`  field of the Jobs API 2.1.

**Comment by allebacco** [comment]
Thanks. Is there any timeline for version 0.4.0?

**Comment by renardeinside** [comment]
I'll release this bug fix today as 0.3.2 - it's relatively small change, upgrade to Jobs API 2.1 shall be a separate PR/issue.

**Comment by renardeinside** [comment]
released in 0.3.2
